tools/change_etype_to_canonical_etype.py

```diff
@@ -41,71 +46,81 @@ def convert_conf(part_config):
-        )
-        canonical_etypes.update(c_etype)
-    return canonical_etypes
+def etype2canonical_etype(part_config, config):
+    try:
+        num_parts = config["num_parts"]
+        edge_map = config[edge_map_key]
+        etypes = list(edge_map.keys())
+        # get part id for each seed edges
+        partition_ids = []
+        for _, bound in edge_map.items():
+            for i in range(num_parts):
+                if bound[i][1] > bound[i][0]:
+                    partition_ids.append(i)
+                    break
+        partition_ids = torch.tensor(partition_ids)
+
+        # start index of each partition
+        shifts = []
+        for i in range(num_parts):
+            shifts.append(edge_map[etypes[0]][i][0])
+        shifts = torch.tensor(shifts)
+
+        canonical_etypes = {}
+        part_ids = [
+            part_id for part_id in range(num_parts) if part_id in partition_ids
+        ]
+        for part_id in part_ids:
+            seed_etypes = [
+                etypes[i]
+                for i in range(len(etypes))
+                if partition_ids[i] == part_id
+            ]
+            c_etype = _find_c_etypes_in_partition(
+                part_id,
+                seed_etypes,
+                config[etypes_key],
+                config["ntypes"],
+                edge_map,
+                shifts,
+                part_config,
+            )
+            canonical_etypes.update(c_etype)
+        return canonical_etypes
+    except ValueError as e:
+        print(e)
 
 
 def _find_c_etypes_in_partition(
-    seed_edges, seed_edge_tids, part_id, part_config
+    part_id, seed_etypes, etypes, ntypes, edge_map, shifts, config_path
 ):
-    folder = os.path.dirname(os.path.realpath(part_config))
-    partition_book = {}
-    local_g = dgl.DGLGraph()
     try:
+        folder = os.path.dirname(os.path.realpath(config_path))
         local_g = load_graphs(f"{folder}/part{part_id}/graph.dgl")[0][0]
-        partition_book = load_partition_book(
-            part_config=part_config, part_id=part_id
-        )[0]
+        local_eids = [
+            edge_map[etype][part_id][0] - shifts[part_id]
+            for etype in seed_etypes
+        ]
+        local_eids = toindex(torch.tensor(local_eids))
+        local_eids = local_eids.tousertensor()
+        local_src, local_dst = local_g.find_edges(local_eids)
+        src_ntids, dst_ntids = (
+            local_g.ndata[dgl.NTYPE][local_src],
+            local_g.ndata[dgl.NTYPE][local_dst],
+        )
+        ntypes = {v: k for k, v in ntypes.items()}
+        src_ntypes = [ntypes[ntid.item()] for ntid in src_ntids]
+        dst_ntypes = [ntypes[ntid.item()] for ntid in dst_ntids]
+        c_etypes = list(zip(src_ntypes, seed_etypes, dst_ntypes))
+        c_etypes = [
+            canonical_etypes_delimiter.join(c_etype) for c_etype in c_etypes
+        ]
+        return {k: etypes[v] for (k, v) in zip(c_etypes, seed_etypes)}
     except DGLError as e:
```

Comment:
which call above may throw `DGLError`? one possible exception could be `dgl.NTYPE` no exist in `local_g.ndata`. can such exception be caught then?

Reply:
Possible error is ```graph.dgl``` not found.
```dgl.NTYP``` can't be found in homogeneous graph, but it will be handled in different branch(see Line 27 ```if statement```).